From 39c440b3b7519d8cdf78655e0eeffe00505b5b04 Mon Sep 17 00:00:00 2001
From: HassanMirza01 <hassanmirza01@hotmail.com>
Date: Tue, 20 Feb 2018 11:14:59 -0500
Subject: [PATCH] "0002-init-vendor-stubs-boot-critical
 _______________________________

*Previous From 1bc87b588a5565f4678fc29244b6e5e0c51b6592 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Sat, 3 Feb 2018 07:34:26 -0600
*Previous Subject: [PATCH] "0002-init-vendor-stubs-boot-critical
*_______________________________
*
**Previous From e2b9e138c2210dd0584d95a4453ba346bb5b034d Mon Sep 17 00:00:00 2001
**Previous From: Meticulus <theonejohnnyd@gmail.com>
**Previous Date: Tue, 12 Dec 2017 05:07:26 -0600
**Previous Subject: [PATCH] "0002-init-vendor-stubs-boot-critical
**_______________________________
**
***Previous From e58d8afd9f67e3c25c19a1ce80523c1168a6525d Mon Sep 17 00:00:00 2001
***Previous From: Meticulus <theonejohnnyd@gmail.com>
***Previous Date: Fri, 1 Dec 2017 16:13:30 -0600
***Previous Subject: [PATCH] "0002-init-vendor-stubs-boot-critical
***_______________________________
***
****Previous From 8cc098b8871224f85f38cf23a38efd8e1dbda7c4 Mon Sep 17 00:00:00 2001
****Previous From: Meticulus <theonejohnnyd@gmail.com>
****Previous Date: Sun, 30 Apr 2017 10:44:21 -0500
****Previous Subject: [PATCH] 0002-init-vendor-stubs-boot-critical
****"
***
***Change-Id: I25f937ec8f7683f833a4d366579d5a7add154987""
*
*Change-Id: If6ceed7dbffc4b55071793be84073eb028122902"

Change-Id: Iac9c71e395fe341f1779e00eb32e46ad0b465d78
---
 init/property_service.cpp |  5 ++++-
 init/vendor_init.cpp      | 12 +++++++++++-
 init/vendor_init.h        |  4 +++-
 3 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index 5f34a16..7a1993c 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -682,6 +682,8 @@ void property_load_boot_defaults() {
     load_properties_from_file("/vendor/default.prop", NULL);
 
     update_sys_usb_config();
+
+    vendor_load_default_properties();
 }
 
 static void load_override_properties() {
@@ -699,6 +701,7 @@ void load_persist_props(void) {
     load_override_properties();
     /* Read persistent properties after all default values have been loaded. */
     load_persistent_properties();
+    vendor_load_persist_properties();
     property_set("ro.persistent_properties.ready", "true");
 }
 
@@ -740,7 +743,7 @@ void load_system_props() {
     load_properties_from_file("/factory/factory.prop", "ro.*");
 
     // Update with vendor-specific property runtime overrides
-    vendor_load_properties();
+    vendor_load_system_properties();
 
     load_recovery_id_prop();
 }
diff --git a/init/vendor_init.cpp b/init/vendor_init.cpp
index d3fd5ff..a30c2f1 100644
--- a/init/vendor_init.cpp
+++ b/init/vendor_init.cpp
@@ -32,6 +32,16 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 /* init vendor override stubs */
 
 __attribute__ ((weak))
-void vendor_load_properties()
+void vendor_load_default_properties()
+{
+}
+
+__attribute__ ((weak))
+void vendor_load_system_properties()
+{
+}
+
+__attribute__ ((weak))
+void vendor_load_persist_properties()
 {
 }
diff --git a/init/vendor_init.h b/init/vendor_init.h
index 9afb449..f86c65f 100644
--- a/init/vendor_init.h
+++ b/init/vendor_init.h
@@ -29,5 +29,7 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #ifndef __INIT_VENDOR__H__
 #define __INIT_VENDOR__H__
-extern void vendor_load_properties(void);
+extern void vendor_load_default_properties(void);
+extern void vendor_load_system_properties(void);
+extern void vendor_load_persist_properties(void);
 #endif /* __INIT_VENDOR__H__ */
-- 
2.14.1

